database/postgresql/src/main/java/com/example/javaopencommerce/catalog/PsqlItemRepositoryImpl.java

```diff
@@ -3,10 +3,12 @@
 
 import com.example.javaopencommerce.exception.EntityNotFoundException;
 
+import javax.enterprise.context.ApplicationScoped;
 import javax.persistence.EntityManager;
 import javax.persistence.NoResultException;
 import java.util.List;
 
+@ApplicationScoped
```

Comment:
Not a fan of that annotation. It would be better to create bean with a @Producer.